grudge/execution.py

```diff
@@ -73,7 +73,15 @@ def map_ones(self, expr):
 
     def map_node_coordinate_component(self, expr):
         discr = self.dcoll.discr_from_dd(expr.dd)
-        return thaw(self.array_context, discr.nodes()[expr.axis])
+        return thaw(self.array_context, discr.nodes(
+            # only save volume nodes or boundary nodes
+            # (but not interior ones, which are likely only used once
+            # to compute the normals)
```

Comment:
Can you elaborate on what you mean by "interior" nodes?

Reply:
How's 85c5c1a?

Reply:
I'm guessing this is good to resolve, too?